### PR TITLE
GROOVY-9060: Make the initial capacity of LinkedHashMap be power of 2…

### PR DESCRIPTION
**Pull request by danielsun1106**
… for better performance

### CODE REVIEW

File: src/main/java/org/codehaus/groovy/runtime/InvokerHelper.java [hunk 1]
```diff
@@ -396,10 +402,33 @@ public static List createList(Object[] values) {
...
+     * 7: 8
+     * 8: 8
+     * 9: 16
+     * ...
+     *
+     * @param initialEntryCnt the initial entry count
+     * @return the initial capacity
+     */
+    public static int initialCapacity(int initialEntryCnt) {
+        return (int) pow(2, ceil(log(initialEntryCnt) / LN2));
```
Comment (blackdrag) [inline src/main/java/org/codehaus/groovy/runtime/InvokerHelper.java]: How about  return 2<<Integer.highestOneBit​(initialEntryCnt)+1; ?

Comment (danielsun1106) [reply src/main/java/org/codehaus/groovy/runtime/InvokerHelper.java]: `2<<Integer.highestOneBit​(initialEntryCnt)+1; ` increases its result too fast, e.g. 
When `initialEntryCnt` is 16, the result has already been 262144 😲

Thanks for your suggestion all the same.

Comment (blackdrag) [reply src/main/java/org/codehaus/groovy/runtime/InvokerHelper.java]: ups, I did misread the documentation. Integer.highestOneBit​(initialEntryCnt)*2 then

Comment (danielsun1106) [reply src/main/java/org/codehaus/groovy/runtime/InvokerHelper.java]: I will tweak the current implementation according to suggestion, which will be more efficient 😄

**Comment by eric-milles** [comment]
Doesn't this just move the math from inside of HashMap to InvokerHelper.initialCapacity?  Where is the performance gain?  Is there any benchmark for this?

Did the constant {{LN2}} get used anywhere?  It appears to be added but unused.

**Comment by danielsun1106** [comment]
See  https://github.com/apache/groovy/commit/5c8f08633a320be8836a249d577a549fe5966b76

**Pull request closed without merging**